### PR TITLE
Feature/refactoring

### LINKED ISSUE #10: Once fetch is working on all three data sets, delete the hardcoded data file
Delete hardcoded data file! woohoo fetch is working

### LINKED ISSUE #45: Fix ingredient search bug
Should be able to search "wheat" or "flour" and see results for "wheat flour" ingredient

### PR DESCRIPTION
**Pull request by clairefields15**
# Description
Added filterByNameOrIngredient to recipe repository and user classes to prevent double pork chop (pork chop name and pork chop ingredient).
Probably totally fucked up our tests passing in the process but it works. Will clean up tests later.
Refactored scripts a lot, removed user names on favorites page and queue and replaced with "your" because we were seeing names ending in 's' having `apostrophe s` added (ex: Viola Davis's favorites).
Fixed some other small bugs.
closes #10 
closes #45

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (clean up code)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -70,9 +70,7 @@ const tags = {
 window.addEventListener('load', getDataFromAPI);
 
 const allCourseButtons = document.querySelectorAll('button').forEach(button =>
```
Comment (Kristiannmiller) [inline src/scripts.js:72]: Nice use of querySelectorAll!

File: src/scripts.js [hunk 2]
```diff
@@ -165,42 +172,34 @@ function populateFavoritesPage(someFavorites) {
...
-  const searchByIngredient = user.favsByIngredient;
-  const searchAll = [searchByName, searchByIngredient];
-  searchAll.forEach(type => {
-    type.forEach(recipe => {
-      favoritesGrid.innerHTML += `
+  const filteredFavs = user.favsByNameOrIngredient;
+  filteredFavs.forEach(recipe => {
+    favoritesGrid.innerHTML += `
         <article id="${recipe.id}" class="mini-recipe-card recipe-target">
           <img class="mini-recipe-img" alt="Picture of ${recipe.name}" src="${recipe.image}">
```
Comment (Kristiannmiller) [inline src/scripts.js:179]: No need to put "image of..." or "picture of..." in the alt text attribute. It's already assumed your alt text is referring to an image, so there's no need to specify it

File: src/scripts.js [hunk 3]
```diff
@@ -165,42 +172,34 @@ function populateFavoritesPage(someFavorites) {
...
-  const searchByIngredient = someRepository.filteredByIngredient;
-  const searchAll = [searchByName, searchByIngredient];
-  searchAll.forEach(type => {
-    type.forEach(recipe => {
-      searchResultGrid.innerHTML += `
+  const filteredRecipes = someRepository.filteredByNameOrIngredient;
+  filteredRecipes.forEach(recipe => {
+    searchResultGrid.innerHTML += `
         <article id="${recipe.id}" class="mini-recipe-card recipe-target">
           <img class="mini-recipe-img" alt="Picture of ${recipe.name}" src="${recipe.image}">
```
Comment (Kristiannmiller) [inline src/scripts.js:192]: No "picture of" necessary

File: src/scripts.js [hunk 4]
```diff
@@ -165,42 +172,34 @@ function populateFavoritesPage(someFavorites) {
...
       `;
-    });
   });
 }
 
 
 //hide and show DOM functions
 function hide(elements) {
- elements.forEach(element => element.classList.add('hidden'));
+  elements.forEach(element => element.classList.add('hidden'));
```
Comment (Kristiannmiller) [inline src/scripts.js:202]: Love these short and sweet show/hide functions!

File: src/scripts.js [hunk 5]
```diff
@@ -274,14 +268,6 @@ function showHeart(foundRecipe) {
...
-  if (id.charAt(id.length - 1) === 's') {
-    browseHeader.innerText = `Browse ${id}`;
-  } else {
-    browseHeader.innerText = `Browse ${id} recipes`;
-  }
-}
-
 function filterByTags(button) {
   changeHeaderText(button.id);
   if (button.id === 'appetizers') {
```
Comment (Kristiannmiller) [inline src/scripts.js:273]: Consider using bracket notation here to cut down on these conditionals

**Review by Kristiannmiller — APPROVED** [review]
This looks great! I left a few notes on ways to dry some of the longer functions up. This is definitely a tough, tedious project and you executed it really well. Let me know if you have any questions.

**Review by darlaevans2000 — APPROVED** [review]
Will merge and then go over code review from Kristi

**Pull request merged**